trestle/core/commands/author/docs.py

```diff
@@ -197,19 +211,49 @@ def _validate_dir(
                     else:
                         logger.info(f'VALID: {self.rel_dir(item_path)}')
                 elif recurse:
-                    if not self._validate_dir(
-                            template_file, governed_heading, item_path, validate_header, validate_only_header, recurse):
+                    if not self._validate_dir(template_file,
+                                              governed_heading,
+                                              item_path,
+                                              validate_header,
+                                              validate_only_header,
+                                              recurse,
+                                              readme_validate):
                         status = 1
         return status
 
-    def validate(self, governed_heading: str, validate_header: bool, validate_only_header: bool, recurse: bool) -> int:
-        """Validate task."""
+    def validate(
+        self,
+        governed_heading: str,
+        validate_header: bool,
+        validate_only_header: bool,
+        recurse: bool,
+        readme_validate: bool
+    ) -> int:
+        """
+        Validate task.
+
+        Args:
+            governed_heading: A heading for which structural enforcement (see online docs).
+            validate_header: Whether or not to validate the key structure of the yaml header to the markdown document.
+            validate_only_header: Whether to validate just the yaml header.
+            recurse: Whether to allow validated files to be in a directory tree.
+            readme_validate: Whether to validate readme files, otherwise they will be ignored.
+
+        Returns:
+            Return code to be used for the command.
+        """
         if not self.task_path.is_dir():
             logger.error(f'Task directory {self.rel_dir(self.task_path)} does not exist. Exiting validate.')
         template_file = self.template_dir / self.template_name
         if not template_file.is_file():
             logger.error(f'Required template file: {self.rel_dir(template_file)} does not exist. Exiting.')
             return 1
         return self._validate_dir(
-            template_file, governed_heading, self.task_path, validate_header, validate_only_header, recurse
+            template_file,
+            governed_heading,
+            self.task_path,
+            validate_header,
+            validate_only_header,
+            recurse,
+            readme_validate
```

Comment:
Just a comment that these multiple options may be better as a separate class or something.  I had added the validate_header and validate_only_header - which overlap in meaning.  And now with readme_validate there is even more.  Will this possibly have more added?  I think it's fine as it is, but does stand out in the code as a lot of arguments in a call.

Reply:
I thought about that. I think we should clean up the functionality a bit going forward. validate header only can technically be dropped due to the separate command.